src/test/java/com/github/dikhan/pagerduty/client/events/domain/PayloadBuilderTest.java

```diff
@@ -22,20 +23,13 @@ public void notAbleToCreatePayloadIfMandatoryFieldSeverityIsMissing() {
     @Test
     public void successfulCreation() {
         Payload.Builder.newBuilder().setSummary("summary").setSource("source")
-                .setSeverity(Severity.INFO).setCustomDetails("details").build();
-    }
-
-    @Test(expected = IllegalArgumentException.class)
-    public void testCustomeDetailInvalidJsonFormat() {
-        String details = "{\"Name\":\"hello\",}";
-
-        Payload.Builder.newBuilder().setSummary("summary").setSource("source")
-                .setSeverity(Severity.INFO).setCustomDetails(details).build();
+                .setSeverity(Severity.INFO).build();
     }
 
     @Test
     public void testCustomeDetailWithValidJson() {
-        String details = "{\"Name\":\"Hello\"}";
+        JSONObject details = new JSONObject("{\"Name\":\"Hello\"}");
```

Comment:
I think it's worth also asserting the content of the json:

        Payload p = Payload.Builder.newBuilder().setSummary("summary").setSource("source")
                .setSeverity(Severity.INFO).setCustomDetails(details).build();
        assertThat(details.toString()).isEqualTo(p.getCustomDetails());

This assumes that p.getCustomDetails() returns a string already...as the user would have set previously the custom details having passed in the actual Json object...but the builder internally converts that into a string simplifying the serialization process and the number of classes needed for that purpose.

Pls let me know if you have any questions.